common/include/common/common_utils.hpp

```diff
@@ -14,6 +14,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <string>
+#include <type_traits>
```

Comment:
Is this still needed by the way ?

Reply:
no, removing it in next commit. Thanks for spotting it!